Modify rule S1784: Migrate to new educational format (#1881)
SonarSource/rspec: rules/S1784/php/rule.adoc

== Why is this an issue?

> Explicitly declaring method visibility provides clarity and improves code readability.
> It makes it clear to other developers how the methods should be accessed and used, enhancing the maintainability of the codebase
> It also promotes the principle of encapsulation, allowing you to control the accessibility of methods and prevent unintended access or modifications from external code.
> By explicitly stating the visibility, you establish a clear contract for how the methods should be interacted with, making it easier to understand and reason about the behavior of the code.

Reviewer: ```suggestion
Explicitly declaring method visibility provides clarity and improves code readability. This leads to:

* Simplifying access to and use of the methods for other developers, enhancing the maintainability of the codebase.
* Promoting the principle of encapsulation, allowing you to control the accessibility of methods and prevent unintended access or modifications from external code.

By explicitly stating the visibility, you establish a clear contract for how the methods should be interacted with, making it easier to understand and reason about the behavior of the code.
```

Available access modifiers are:

* ``++private++`` - access allowed only within the same class
* ``++protected++`` - access allowed to the class and its child classes
* ``++public++`` - unfettered access by all (default)

> == How to fix it?

Reviewer: ```suggestion
== How to fix it
```
Or in Core PHP
